rs/pocket_ic_server/src/state_api/state.rs

```diff
@@ -614,8 +617,15 @@ async fn handler(
             req.send().await
         };
 
+        let mut http_response = resp.canister_response;
+        if !state.is_https {
+            http_response
+                .headers_mut()
+                .remove(UPGRADE_INSECURE_REQUESTS);
```

Comment:
@adamspofford-dfinity Could you please suggest how the headers should be modified here?

Reply:
- `Upgrade-Insecure-Requests` should not be blocked (or rather since it's a request header it shouldn't appear in the first place)
- `Strict-Transport-Security` should be blocked
- `Content-Security-Policy`, if it exists, should be modified to remove the `upgrade-insecure-requests;` directive, if it exists

Reply:
Modifying the content of the CSP header looks involved to me. I see the following alternatives:
- block CSP entirely for localhost (this would make local testing of CSP impossible though)
- modify the content of the CSP header when deploying an asset canister to the local network
The latter seems to be the most robust option in my opinion.